### ISSUE TITLE
Changes needed to placeholder styling to meet contrast requirements

### ISSUE THREAD

**Issue by zzucco** [wet-boew/wet-boew#8409]
To meet the WCAG requirement 1.4.3 on text contrast, we need to make changes to the way we're styling placeholder text.

**tl;dr** we should either set placeholder text to #767676, #636363 or to #000 + 0.56 opacity.

What follows is my research and analysis of the issue:

First off, the colour of the text depends on the background of the element it is typed on. Most of our input elements are white, but the search bar on canada.ca is #e0e0e0, so these are the two colours against which I'm comparing. (using [contrast-ratio.com](http://contrast-ratio.com/)). Other background colours will not be guaranteed to work.

Each browser has its own default styling for placeholders.

- IE11: #6d6d6d
- Edge: #666666
- Chrome: #757575
- Firefox: #000 with 0.54 opacity

All of these fail to meet contrast requirements.

Currently,  WET uses bootstrap which sets the colour of placeholder text to #999999. This too, fails.

**Here are our options:**

1. Set the placeholder colour to #767676. This is the palest value that has a contrast ratio of >= 4.5 against a white background. If we do this, the GCWeb theme will need to override it.
2. Set the placeholder colour to #636363. This contrasts well against both white and #e0e0e0, so it's a good one-size-fits-all colour, but the darker tone makes it harder to set apart from real text.
3. Set the placeholder colour to #000 and the opacity to 0.56. This will appear as #636363 against #e0e0e0 and #717171 against white.

All three of these would require a simple change to the bootstrap override file.

See [GCWeb PR #1363](https://github.com/wet-boew/GCWeb/pull/1363) and [GCWeb issue #1285](https://github.com/wet-boew/GCWeb/issues/1285).

https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Placeholder_Research

/cc @duboisp

**Comment by duboisp** [comment]
@zzucco can you provide a screen shoot of all options your are proposing.

**Comment by zzucco** [comment]
![default](https://user-images.githubusercontent.com/32529367/42096348-148bcfac-7b83-11e8-81b8-6d5a2aa1010d.PNG)

Here is a comparison of each style. You can see how minor the difference is - this change is really just to satisfy the requirements, it's far too minimal to actually be of use to anyone.

If we want to make our placeholders more noticeable while still satisfying the contrast ratio, we could consider a darker colour for normal text. See the black text in the picture.

**Comment by duboisp** [comment]
@shawnthompson as discussed, I would appreciate if you can share the use case that support to not use placeholder. We can create a documentation page that explain it and then release a design decision which would be a best practice to not use placeholder.

**Comment by shawnthompson** [comment]
I already started a discussion in the Design System to advise people not to use placeholder attributes. 

https://github.com/gctools-outilsgc/design-system/issues/181#issuecomment-406639154

I've also started a chat within the a11y GoC community.

Other articles that talk about it.

* https://www.digitala11y.com/anatomy-of-accessible-forms-placeholder-is-a-mirage/

There is also this researching from the W3C on issues for low vision users.
* https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Placeholder_Research

Here is part of why it was developed to help cognitive users.
* https://www.w3.org/WAI/GL/wiki/Using_@placeholder_on_input

This link talks about how it can be harmful and how easy it is to go wrong.
* https://www.nngroup.com/articles/form-design-placeholders/

More details on placeholder research
* https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Placeholder_Research

4 Great Reasons why not to user placeholder text
* https://medium.theuxblog.com/4-reasons-to-avoid-using-placeholder-text-in-forms-8fc33e724e4f

If you are going to go ahead and use a placeholder attribute this goes over how to best do it
* https://a11yproject.com/posts/placeholder-input-elements/

cc/ @jrowsell @andrewnordlund @EricDunsworth
